src/index.ts

```diff
@@ -1,387 +1,431 @@
-import amqp from 'amqplib';
-import { ConsumeMessage, Connection, Channel, Options } from 'amqplib';
-import { Logger } from 'log4js';
+import amqp, {
+    ChannelWrapper,
+    AmqpConnectionManager,
+} from 'amqp-connection-manager';
+import {ConsumeMessage, Channel, Options} from 'amqplib';
+import {Logger} from 'log4js';
 import MessageBatchingManager from './helpers/message_batching_manager';
 
-export { ConsumeMessage, Channel };
+type ConnectCallback = (channel: Channel) => Promise<any>;
+
+export {ConsumeMessage, ChannelWrapper, Channel, ConnectCallback};
 
 const DEFAULT_MAX_FILES_SIZE_BYTES = 1024 * 1024 * 2; // 2 MB
 const DEFAULT_MAX_BUFFER_TIME_MS = 60 * 1000; // 60 seconds
 
 // Used as return value in handler for registerConsumerBatch function
 export interface ConsumeBatchMessages {
-  batchingOptions: {
-    maxSizeBytes?: number;
-    maxTimeMs?: number;
-  };
-  totalSizeInBytes: number;
-  messages: Array<ConsumeMessage>;
-  ackAll: (allUpTo?: boolean) => void;
-  nackAll: (allUpTo?: boolean, requeue?: boolean) => void;
+    batchingOptions: {
+        maxSizeBytes?: number;
+        maxTimeMs?: number;
+    };
+    totalSizeInBytes: number;
+    messages: Array<ConsumeMessage>;
+    ackAll: (allUpTo?: boolean) => void;
+    nackAll: (allUpTo?: boolean, requeue?: boolean) => void;
 }
 
 // Used for registerConsumer function
-export interface ConsumerOptions extends Options.Consume {}
+export interface ConsumerOptions extends Options.Consume {
+}
 
 // Used for registerConsumerBatch function
 export interface ConsumerBatchOptions extends Options.Consume {
-  batching?: {
-    maxSizeBytes?: number;
-    maxTimeMs?: number;
-  };
+    batching?: {
+        maxSizeBytes?: number;
+        maxTimeMs?: number;
+    };
 }
 
 export interface IAmqpCacoonConfig {
-  protocol?: string;
-  username?: string;
-  password?: string;
-  host?: string;
-  port?: number;
-  connectionString?: string;
-  amqp_opts: object;
-  providers: {
-    logger?: Logger;
-  };
-  maxWaitForDrainMs?: number; // How long to wait for a drain event if RabbitMq fills up. Zero to wait forever. Defaults to 60000 ms (1 min)
+    protocol?: string;
+    username?: string;
+    password?: string;
+    host?: string;
+    port?: number;
+    connectionString?: string;
+    amqp_opts: object;
+    providers: {
+        logger?: Logger;
+    };
+    onChannelConnect?: ConnectCallback;
+    onBrokerConnect?: Function;
```

Comment:
Is there are reason we are using `Function` instead of something like the following?
```
onBrokerConnect?: () => void
// Or if you really need it to be any input and any output
onBrokerConnect?: (...args: any[]) => any
```

Reply:
Oh just probably my own JS / TS clumsiness!

Reply:
updated